File: server/Models/Interest.cs


namespace Models
{
    public class Interest
    {
        public int InterestId { get; set; }
        public string StudentId { get; set; }
        public int BandId { get; set; }
        public DateTime InterestDate { get; set; }

        // Navigation Properties
        public Student Student { get; set; }
        public Band Band { get; set; }
    }
}

File: client/src/models/Interest.ts
import Band  from './Band';
import Student  from './Student';

export default interface Interest {
    interestId: number;
    studentId: string; // Reference to the interested student
    bandId: number; // Reference to the band of interest
    isInterested: boolean;
    interestDate: string;
    student?: Student;
    band?: Band;
}
